cmd/devicesimulator/main.go

```diff
@@ -233,101 +288,167 @@ func createAgentConfigTemplate(dataDir string, configFile string) *agent_config.
 		log.Fatalf("Error validating config: %v", err)
 	}
 
-	return agentConfigTemplate
+	// if the config does not have an explicit log level we can override it
+	return agentConfigTemplate, tmpConfig.LogLevel == ""
 }
 
-func createAgents(log *logrus.Logger, numDevices int, initialDeviceIndex int, agentConfigTemplate *agent_config.Config) ([]*agent.Agent, []string) {
-	log.Infoln("creating agents")
-	agents := make([]*agent.Agent, numDevices)
-	agentsFolders := make([]string, numDevices)
+func copyAgentFiles(log *logrus.Logger, certDir, agentDir string) {
+	copyDir := func(path string) string {
+		return filepath.Join(agentDir, path)
+	}
+	fileCopyMap := map[string]struct {
+		dir   string
+		files []string
+	}{
+		certDir: {dir: copyDir(agent_config.DefaultConfigDir), files: []string{"ca.crt", "client-enrollment.crt", "client-enrollment.key"}},
```

Comment:
Since this is a device simulator, maybe it is better for it not to be aware of "real" network configurations?

Reply:
I tried supplying it with a well formatted routing table that contained faked data, but the default route finding logic matches the routing table against network interfaces that it discovers via golangs `net.Interfaces`. So unless the fake table matches interfaces locally, it'll still take 45 seconds to resolve nothing.

As the network info is just used for reporting purposes, I do think it would be preferable to be able to disable collecting the network info, but that is a more involved agent change.

Reply:
BTW, agent uses its own reader and writer.  Each of them may have some pre-set prefix as part of the file.
With this you can use it to use some your own fake data.
Is the 45 seconds accumulative or the whole process is delayed only by  this duration?

Reply:
I implemented the fix in the agent to not make this required anymore: #1526